crates/sui-types/src/move_package.rs

```diff
@@ -136,9 +230,59 @@ impl MovePackage {
                 })
                 .collect(),
             max_move_package_size,
+            type_origin_table,
+            dependencies,
         )
     }
 
+    fn build_linkage_table(
+        dependencies: &[MovePackage],
+    ) -> Result<BTreeMap<ObjectID, UpgradeInfo>, ExecutionError> {
+        // all transitive dependencies are included so simply put them into the table
+
+        // TODO: there may be a better Rust incantation for this but dealing with error mapping in
+        // closures is painful sometimes
+        let mut linkage_table = BTreeMap::new();
+        for p in dependencies {
+            let bytes: &Vec<u8> =
+                p.module_map.values().next().expect(
+                    "Tried to build a Move package from an empty iterator of Compiled modules",
+                );
+            let module = CompiledModule::deserialize(bytes)
+                .map_err(|_| ExecutionErrorKind::InvalidMoveModule)?;
+            let module_handle = module.module_handle_at(module.self_handle_idx());
+            let original_id: ObjectID =
+                (*module.address_identifier_at(module_handle.address)).into();
```

Comment:
`module` should expose an `address()` function (I think it's part of the `ModuleAccess` trait) that should simplify this for you.